### ISSUE TITLE
Take the 3.1 release branch and merge it sans code onto master and deletethe branch.

### ISSUE THREAD

**Issue by hazendaz** [spotbugs/spotbugs#2776]

**Comment by hazendaz** [comment]
Keeping branches around do us no good.  We can merge now entirely complete 3.1 back to master so the commits show and we can always cut back out but without it affecting our code as it is now.  This will allow that branch to be then deleted.  I'll take care of that then I'll further reapply the one single branch on top of that to on top our current master provided it was not maybe already done.  Just adding this as note to do so.  It won't affect our usage but is house cleaning we need.

**Comment by JuditKnoll** [comment]
There are some commits and the functionality they provide, which are in the release-3.1 branch, but are missing from master (see the [compare](https://github.com/spotbugs/spotbugs/compare/master...release-3.1)).
I was looking at issue https://github.com/spotbugs/spotbugs/issues/705 and the PR https://github.com/spotbugs/spotbugs/pull/797 implementing it in particular, when stumbled upon this problem. I couldn't find the commit removing this detector, so it looks like this was not even introduced to 4.0.0.
@KengoTODA @iloveeclipse Do you remember, was there a reason for leaving these out, was it intentional or was it only some merge issue?

**Comment by iloveeclipse** [comment]
> Do you remember, was there a reason for leaving these out, was it intentional or was it only some merge issue?

I honestly can't remember what that all mean and why it wasn't merged.

**Comment by iloveeclipse** [comment]
> I honestly can't remember what that all mean and why it wasn't merged.

Probably the work was superceeded by changes on 4.x branch? Wasn't the time where we switched to gradle and so on?

**Comment by JuditKnoll** [comment]
> > I honestly can't remember what that all mean and why it wasn't merged.
> 
> Probably the work was superceeded by changes on 4.x branch? Wasn't the time where we switched to gradle and so on?

Yes, I think so, this was the same time. Some changes got left out, which I think would be useful. Maybe the mixup or at least some of it was because of https://github.com/spotbugs/spotbugs/pull/850.
